### ISSUE TITLE
[Build Error][Windows] "build.cmd" cannot run for second time due to "mkdir %cmake-root%" will always fail.

### ISSUE THREAD

**Issue by yphuangms** [Azure/azure-iot-gateway-sdk#54]
This "build.cmd" contains some scripts that result in conflicts result at following lines:

1. build script will remove "build" folder at first

        rmdir /s/q %cmake-root%

2. build script call git command to clean submodule folder

        git submodule foreach --recursive --quiet "rm -rf build/"
    
     by doing so, it somehow create "build" folder again under %build-root%, this is the main cause that fail the following step.

3. create "build" folder and assert for the command success, this always fail due to the command in step 2.

      mkdir %cmake-root%
      if not !ERRORLEVEL!==0 exit /b !ERRORLEVEL!

We also found that "%build-root%/build" folder is not the only folder to keep VS files and temporary build outputs. Some of the files are inside submodule folder, and some are not. For example, submodule "nanomsg" will have the VS files and build outputs at the same folder of the submodule source code, while submodule "parson" will have the VS files and build outputs under ""%build-root%/build". Because running "git submodule foreach --recursive --quite 'rm -rf build'" takes a lot of time in the build process, if we keep the build outputs all under "%build-root%/build", then there's no need to traverse into all submodule source tree to do clean-up, and this will also prevent from "build" folder being created to block the success call to mkdir %cmake-root%.

**Comment by andrew-buckley** [comment]
I was able to run the build.cmd script multiple times successfully. Please provide repro steps if possible.

Thanks!

**Comment by damonbarry** [comment]
Unable to reproduce, and haven't seen any other reports of this issue, so I'm closing it. Please reopen if you can provide steps to reproduce. Thanks!

**Comment by yphuangms** [comment]
The problem still exists in the release of "2016-12-16". After a few more repeated tests, we found the problem is not 100% (always), but it occurred often in my development environment (Windows 8.1 + Visual Studio 2015 Update2). 

In previous message I suspected "git submodule foreach ..." is the command that re-generates "build" folder, since it's the only command executed in between "rmdir" and "mkdir". But I was wrong. This time I monitor the root folder, and when the "build" folder is created during "git submodule foreach ...", this "build" folder contains the following files:

 Directory of  D:\github\azure-iot-gateway-sdk.1216\build\CMakeFiles\3.6.1\CompilerIdC\Debug\CompilerIdC.tlog

12/19/2016  16:30    <DIR>          .
12/19/2016  16:30    <DIR>          ..
12/19/2016  16:19               680 link-VCTIP-wermgr.read.1.tlog
12/19/2016  16:19               686 link-VCTIP-wermgr.read.2.tlog
12/19/2016  16:19               754 link-VCTIP-wermgr.write.1.tlog
12/19/2016  16:30               524 link-VCTIP.delete.28.tlog
12/19/2016  16:14               524 link-VCTIP.delete.36.tlog
12/19/2016  16:20               524 link-VCTIP.delete.46.tlog
12/19/2016  16:29               524 link-VCTIP.delete.85.tlog
12/19/2016  16:30               524 link-VCTIP.delete.96.tlog
12/19/2016  16:30             6,100 link-VCTIP.read.1.tlog
12/19/2016  16:14             5,644 link-VCTIP.read.11.tlog
12/19/2016  16:14               562 link-VCTIP.read.14.tlog
12/19/2016  16:14             2,584 link-VCTIP.read.15.tlog
12/19/2016  16:14             1,164 link-VCTIP.read.17.tlog
12/19/2016  16:14               414 link-VCTIP.read.24.tlog
12/19/2016  16:30               364 link-VCTIP.read.28.tlog
12/19/2016  16:30               232 link-VCTIP.read.29.tlog
12/19/2016  16:30               232 link-VCTIP.read.31.tlog
12/19/2016  16:14               232 link-VCTIP.read.36.tlog
12/19/2016  16:19            17,748 link-VCTIP.read.40.tlog
12/19/2016  16:19               348 link-VCTIP.read.44.tlog
12/19/2016  16:20               232 link-VCTIP.read.46.tlog
12/19/2016  16:19               348 link-VCTIP.read.49.tlog
12/19/2016  16:24               348 link-VCTIP.read.66.tlog
12/19/2016  16:24               348 link-VCTIP.read.68.tlog
12/19/2016  16:29               348 link-VCTIP.read.80.tlog
12/19/2016  16:29               232 link-VCTIP.read.85.tlog
12/19/2016  16:14               348 link-VCTIP.read.9.tlog
12/19/2016  16:30               232 link-VCTIP.read.96.tlog
12/19/2016  16:30               232 link-VCTIP.write.1.tlog
12/19/2016  16:30               232 link-VCTIP.write.28.tlog
12/19/2016  16:14               232 link-VCTIP.write.36.tlog
12/19/2016  16:19               232 link-VCTIP.write.40.tlog
12/19/2016  16:20               232 link-VCTIP.write.46.tlog
12/19/2016  16:29               232 link-VCTIP.write.85.tlog
12/19/2016  16:30               554 link-VCTIP.write.96.tlog
              35 File(s)         44,746 bytes

     Total Files Listed:
              35 File(s)         44,746 bytes

VCTIP.exe is "Microsoft VC compiler and tools experience improvement data uploader", and it seemed to trigger some logging under "build" folder, and the interval of this automatic trigger seemed to be every 5 minutes, for example, 3:45, 3:50, and will last for 1 to 2 minutes, and if to run "build.cmd" during this period of time, the removed "build" folder will be re-generated and hence cause "mkdir" of "build" folder failed and hence abort "build.cmd".

The suggestion is to disable the check of the success of "mkdir %cmake-root%". Instead, this "build.cmd" could just check if "%cmake-root%" exists right after "mkdir".
